tools/src/launcher/Main.hs

```diff
@@ -559,16 +599,31 @@ system'
     -- ^ Command
     -> [Text]
     -- ^ Lines of standard input
+    -> Executable
+    -- ^ node/wallet log output
     -> io ExitCode
     -- ^ Exit code
-system' phvar p sl = liftIO (do
+system' phvar p sl nt = liftIO (do
     let open = do
-            (m, _, _, ph) <- Process.createProcess p
+            (hIn, stdO, stdE, ph) <- Process.createProcess p
             putMVar phvar ph
-            case m of
-                Just hIn -> IO.hSetBuffering hIn IO.LineBuffering
+            case [hIn,stdO, stdE] of
+                [(Just hndl), (Just o), (Just e)] -> do
+                    -- log certain kind of processes since
+                    -- inherited ones might create infinite loop
+                    case (Process.std_out p, Process.std_err p) of
+                        (Process.CreatePipe, Process.CreatePipe) -> do
+                            _ <- asyncLog o e ph nt
+                            return ()
+                        _ -> return ()
+                    case (Process.std_out p, Process.std_err p) of
+                        (Process.UseHandle _, Process.UseHandle _) -> do
```

Comment:
`UseHandle` means that we want to redirect all console output to a file. Previously `redirect` meant that **instead** of printing to console we were printing to a file. But after this change messages will be printed to a file **and** to console.
First, I think that more natural meaning of redirect is to print only to a file.
Second, I'd like not to change old behaviour.

tl;dr: let's use `asyncLog` only in `CreatePipe` case.